src/main/app/form/validation.test.ts

```diff
@@ -100,31 +100,99 @@ describe('Validation', () => {
   });
 
   describe('isDateInputNotFilled()', () => {
-    test('Should check if date entered is future date', async () => {
+    test('Should return incompleteDayAndMonthAndYear error if no date fields are entered', async () => {
       const date = {
         day: '',
         month: '',
         year: '',
       };
 
       let isValid = isDateInputNotFilled(undefined);
-      expect(isValid).toStrictEqual('invalidDate');
+      expect(isValid).toStrictEqual('incompleteDayAndMonthAndYear');
 
       isValid = isDateInputNotFilled(date);
-      expect(isValid).toStrictEqual('invalidDate');
+      expect(isValid).toStrictEqual('incompleteDayAndMonthAndYear');
+    });
+
+    test('Should return incompleteDayAndMonth error if no day and month fields are entered', async () => {
+      const date = {
+        day: '',
+        month: '',
+        year: '2000',
+      };
+
+      const isValid = isDateInputNotFilled(date);
+      expect(isValid).toStrictEqual('incompleteDayAndMonth');
+    });
+
+    test('Should return incompleteDayAndYear error if no day and year fields are entered', async () => {
+      const date = {
+        day: '',
+        month: '12',
+        year: '',
+      };
+
+      const isValid = isDateInputNotFilled(date);
+      expect(isValid).toStrictEqual('incompleteDayAndYear');
+    });
+
+    test('Should return incompleteMonthAndYear error if no day and year fields are entered', async () => {
+      const date = {
+        day: '31',
+        month: '',
+        year: '',
+      };
+
+      const isValid = isDateInputNotFilled(date);
+      expect(isValid).toStrictEqual('incompleteMonthAndYear');
+    });
+
+    test('Should return incompleteDay error if no day and year fields are entered', async () => {
```

Comment:
The test description looks wrong. Perhaps it should be 'Should return incompleteDay error if no day field is entered'